edgexfoundry/device-sdk-go
# feat!: replace REST device validation callback with MessageBus

BREAKING CHANGE: remove /validate/device REST endpoint

closes #1260 

Signed-off-by: Chris Hung <chris@iotechsys.com>







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  <link to docs PR>

## Testing Instructions

1. Run `metadata` from https://github.com/edgexfoundry/edgex-go/pull/4366 branch
1. Run `device-simple` from this branch with `DEBUG` log
2. Verify logs contains messages like:
```shell
level=INFO ts=2023-02-15T17:11:54.626303Z app=device-simple source=validation.go:29 msg="Subscribing to device validation requests on topic: edgex/device-simple/validate/device"
level=INFO ts=2023-02-15T17:11:54.626311Z app=device-simple source=validation.go:32 msg="Responses to device validation requests will be published on topic: edgex/response/device-simple/<requestId>"
level=DEBUG ts=2023-02-15T17:11:54.650768Z app=device-simple source=validation.go:58 msg="Device validation request received on message queue. Topic: edgex/device-simple/validate/device, Correlation-id: d62b5ef3-df40-47a3-a4c0-f9a33183477b"
level=DEBUG ts=2023-02-15T17:11:54.653205Z app=device-simple source=validation.go:100 msg="Device validation response published on message queue. Topic: edgex/response/device-simple/27f210de-2e8d-464f-9ea9-f00efbb3c219, Correlation-id: d62b5ef3-df40-47a3-a4c0-f9a33183477b"

```

## New Dependency Instructions (If applicable)
